numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -3694,6 +3693,7 @@ _vec_string_with_args(PyArrayObject* char_array, PyArray_Descr* type,
         if (item_result == NULL) {
             goto err;
         }
+        Py_DECREF(args_tuple);
```

Comment:
Shouldn't this happen before the `if`? If not I think we will be leaking a reference if the function call errors.